mlflow/pipelines/steps/predict.py

```diff
@@ -107,6 +107,27 @@ def _run(self, output_directory):
                 error_code=BAD_REQUEST,
             )
 
+        # check if output location is already populated as we allow no overwrites
+        output_format = self.step_config["output_format"]
+        output_location = self.step_config["output_location"]
+        if output_format == "parquet" or output_format == "delta":
+            output_populated = os.path.exists(output_location)
+        else:
```

Comment:
`else if output_format == "spark_sql"` to be more specific on when this code block runs. We could introduce more data format in the future.